### DISCUSSION TITLE
Flask integration using `app.extensions` instead of `app.config`

### DISCUSSION THREAD

**Discussion by alexrudy** [hynek/svcs#71] [Ideas]
This is a minor nit, but I find that `app.config` is usually filled with things which might be set from an external `config.py` file, environment variable, or some other source. I often even like to be able to render these things back to e.g. JSON to save the current configuration for debugging.

OTOH, flask provides [`app.extensions`](https://flask.palletsprojects.com/en/3.0.x/api/#flask.Flask.extensions) as a specific place for extensions to store state, and many flask extensions do that (e.g. Flask-SQLAlchemy).

What do you think about having `svcs` store the `Registry` object (which is managed by `svcs`, and wouldn't make much sense to set as the end user) in `app.extensions`?

**Comment by hynek** [comment]
Oh, I have to admit I didn’t know about app.extension!

**Comment by hynek** [comment]
Looks like it doesn’t link back, so here’s the issue: https://github.com/hynek/svcs/issues/72

**Reply by alexrudy** [reply]
Awesome, thanks! I've made a PR - #73

Weird that GitHub makes it easy to flow from issues to discussions, but the other direction feels a bit half-baked.

**Reply by hynek** [reply]
yeah it's a bit awkward. auto-closing doesn't work either. hope they improve it because the separation of issues and discussions makes a lot of sense.

**Comment by hynek** [comment]
fixed by #73
